### ISSUE TITLE
asyncserver doesn't start on python3.10: "TypeError: As of 3.10, the *loop* parameter was removed from Lock() since it is no longer necessary"

### ISSUE THREAD

**Issue by afanasevps** [grpc/grpc#27888]
### What version of gRPC and what language are you using?
 grpcio-1.41.1,  grpcio-1.41.0

### What operating system (Linux, Windows,...) and version?

1. Linux 5.11.0-38-generic # 42~20.04.1-Ubuntu SMP Tue Sep 28 20:41:07 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
2. macOS 11.6

### What runtime / compiler are you using (e.g. python version or version of gcc)
python 3.10

### What did you do?
try start server under python3.10, install lates version from python.org

### What did you expect to see?
running grpc server

### What did you see instead?
```
Traceback (most recent call last):
  File "/Users/pavelafanasev/work/proto/test.py", line 132, in <module>
    asyncio.run(grpc_serve([DecisionMakerInterface]))
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/base_events.py", line 641, in run_until_complete
    return future.result()
  File "/Users/pavelafanasev/work/proto/test.py", line 115, in grpc_serve
    rpc_server = aio.server()
  File "/Users/pavelafanasev/work/.venv10/lib/python3.10/site-packages/grpc/aio/_server.py", line 207, in server
    return Server(migration_thread_pool, () if handlers is None else handlers,
  File "/Users/pavelafanasev/work/.venv10/lib/python3.10/site-packages/grpc/aio/_server.py", line 54, in __init__
    self._server = cygrpc.AioServer(
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi", line 857, in grpc._cython.cygrpc.AioServer.__init__
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/locks.py", line 77, in __init__
    super().__init__(loop=loop)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/mixins.py", line 17, in __init__
    raise TypeError(
TypeError: As of 3.10, the *loop* parameter was removed from Lock() since it is no longer necessary
```

**Comment by lidizheng** [comment]
@afanasevps Thanks for reporting this. We fixed this issue in #27635, but it's the released with v1.41.x.

**Comment by lidizheng** [comment]
Backport merged. This fix is in `v1.42.0rc1`. I will try to propose a patch to `v1.41.x`, but not sure if it will go through.
